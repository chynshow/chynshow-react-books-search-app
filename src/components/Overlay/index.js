import React from 'react';
import PropType from 'prop-types';

const Overlay = ({ opacity, background, zIndex }) => (
  <div style={{ opacity, background, zIndex }} className='c-overlay' />
);

Overlay.propTypes = {
  opacity: PropType.string.isRequired,
  background: PropType.string.isRequired,
  zIndex: PropType.string,
};

Overlay.defaultProps = {
  opacity: '0',
  background: '#fff',
  zIndex: '0',
};

export default React.memo(Overlay);
