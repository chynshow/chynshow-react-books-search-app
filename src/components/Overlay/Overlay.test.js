import React from 'react';
import Overlay from './';
import { shallow } from 'enzyme';

const setup = (props) => shallow(<Overlay {...props} />);

it('should renders Overlay component with props', () => {
  const c = setup({
    opacity: '1',
    background: '#fff',
    zIndex: '0',
  });
  expect(c).toMatchSnapshot();
});

it('should renders Overlay component without props', () => {
  const c = setup();
  expect(c).toMatchSnapshot();
});
