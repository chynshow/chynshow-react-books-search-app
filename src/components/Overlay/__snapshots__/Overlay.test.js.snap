// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders Overlay component with props 1`] = `
<div
  className="c-overlay"
  style={
    Object {
      "background": "#fff",
      "opacity": "1",
      "zIndex": "0",
    }
  }
/>
`;

exports[`should renders Overlay component without props 1`] = `
<div
  className="c-overlay"
  style={
    Object {
      "background": "#fff",
      "opacity": "0",
      "zIndex": "0",
    }
  }
/>
`;
